### ISSUE TITLE
Proposal: duplicate refer linter

### ISSUE THREAD

**Issue by jramosg** [clj-kondo/clj-kondo#2735]
- [x] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

[ To keep development of this project going, consider sponsoring. If you are
already a sponsor, thank you! You can remove this message. ]

**Is your feature request related to a problem? Please describe.**

Currently, clj-kondo does not warn about duplicate entries in the `:refer` vector of a `require` statement. For example:

```clojure
(ns foo
  (:require [clojure.set :refer [union union]]))

(union #{1} #{2})
```

This is accepted without warning, but the duplicate `union` in `:refer` is redundant and could indicate a copy-paste or editing mistake.

**Describe the solution you'd like**

Add a linter that warns when the same var is listed more than once in a `:refer` vector. This would help catch accidental duplication and keep require statements clean.

**Describe alternatives you've considered**

- Leaving this unchecked (status quo)

**Additional context**

- This is a minor but useful check for code hygiene.
- No warning is currently emitted for this case.

I will do a PR
